safe-global/safe-wallet-web
# fix: Don't allow duplicate signers in safe creation

## What it solves

Resolves #3415 

The fix implemented in #3060 added a new `onChange` call to the `Autocomplete` to render the `AddressInputReadOnly` whenever the user selects an address book entry in the input field and display validation errors in this separate component. 

However if a `deps` array is also passed to `AddressBookInput` it triggers a revalidation which is not propagated because `AddressInputReadOnly` is rendered instead of `AddressInput`.

## How this PR fixes it

- Simplify and move `AddressInputReadOnly` into `AddressInput`
- Add the read-only logic into `AddressInput`
- Adds a new `useChain` hook

## How to test it

1. Create a new Safe
2. Add the same owner twice
3. Observe an error is visible

Make sure that the fix from #3060 is still working:

1. Open an existing safe
2. Go to the owner settings and add a new owner
3. Choose the safe itself as an owner
4. Observe an error is visible

Check that other places where `AddressInput` and `AddressBookInput` are used are still working
- Create address book entry
- Load safe
- TxHistory filter
- Create transactions
- Recovery

## Screenshots
<img width="793" alt="Screenshot 2024-03-12 at 16 10 22" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/c4557df8-3b8e-4248-b2e4-f29c1e58b093">
<img width="701" alt="Screenshot 2024-03-12 at 16 10 43" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/c3a29526-243e-4da0-87c7-3d64e9209335">

## Checklist
